### DISCUSSION TITLE
How to validate Shell navigation in unit tests?

### DISCUSSION THREAD

**Discussion by rthanga1** [dotnet/maui#15841] [Q&A]
In my view model, on a button click, using command, I navigate to a different screen using Shell. 
Like this:
Shell.Current.GoToAsync("//HomePage");

But since Shell.Current is null when I run unit test, the tests crashes and fails. How do I validate Shell naviation in the ViewModel when used with Shell?

**Comment by Saccomani** [comment]
the best way is use a INavigationService and inside call the shell navigation methods, in my case I create a current page property and when I am using the unit test I skip the shell...

**Reply by Saccomani** [reply]
<img width="599" alt="image" src="https://github.com/user-attachments/assets/14e0e353-f635-4739-93e9-d1f92828591a">
something like this
